WcaOnRails/app/models/country.rb

```diff
@@ -2,9 +2,29 @@
 
 class Country < ApplicationRecord
   include Cachable
+  WCA_STATES_JSON_PATH = Rails.root.to_s + "/config/wca-states.json"
   self.table_name = "Countries"
 
-  MULTIPLE_COUNTRIES_IDS = %w(XA XE XS).freeze
+  MULTIPLE_COUNTRIES = [
+    { id: 'XA', name: 'Multiple Countries (Asia)', continentId: '_Asia', iso2: 'XA' },
+    { id: 'XE', name: 'Multiple Countries (Europe)', continentId: '_Europe', iso2: 'XE' },
+    { id: 'XS', name: 'Multiple Countries (South America)', continentId: '_South America', iso2: 'XS' },
+  ].freeze
+
+  MULTIPLE_COUNTRIES_IDS = MULTIPLE_COUNTRIES.map { |c| c[:id] }.freeze
+
+  WCA_STATES = JSON.parse(File.read(WCA_STATES_JSON_PATH)).freeze
+
+  ALL_STATES = [
+    WCA_STATES["states_lists"].map do |list|
+      list["states"].map do |state|
+        state_id = state["id"] || I18n.transliterate(state["name"]).tr("'", "_")
```

Comment:
Interesting question :)
Almost every country's id can be derived from it's name, except for two of them: USA and Korea.
So there is the possibility to provide a custom id for the country in the source markdown file (see [here](https://github.com/thewca/wca-regulations/pull/442/files#diff-d6cffd53d4a7b45a1bb5257bb9a7a743R195).

So here the affectation rather mean "take the custom id, if there is none create it from the name".